jupyter-js-widgets/src/utils.js

```diff
@@ -66,11 +66,18 @@ function loadClass(class_name, module_name, registry) {
 
             // If the module is jupyter-js-widgets, we can just self import.
             var modulePromise;
-            if (module_name === 'jupyter-js-widgets') {
-                modulePromise = Promise.resolve(require('../'));
+            var requireJSDefined = Boolean(typeof window !== 'undefined' && window.requirejs);
```

Comment:
The typeof keyword returns a string with the type name.  `undefined` is the type name of `undefined` (I feel silly typing that).  You can't compare to `undefined` because you'll get a runtime error that the `window` variable hasn't been defined yet.